File: Php/app/Urls/CashgameCheckpointDeleteUrlModel.php
<?php
namespace app\Urls{

	use app\UrlFormatter;
	use app\Urls\BaseClasses\UrlModel;
	use app\RouteFormats;
	use entities\Checkpoints\Checkpoint;
	use entities\Cashgame;
	use entities\Homegame;
	use entities\Player;

	class CashgameCheckpointDeleteUrlModel extends UrlModel{

		public function __construct(Homegame $homegame, Cashgame $cashgame, Player $player, Checkpoint $checkpoint){
			$isoDate = UrlFormatter::formatIsoDate($cashgame->getStartTime());
			$encodedPlayerName = rawurlencode($player->getDisplayName());
			$url = sprintf(RouteFormats::cashgameCheckpointDelete, $homegame->getSlug(), $isoDate, $encodedPlayerName, $checkpoint->getId());
			parent::__construct($url);
		}

	}

}

File: Web/Models/UrlModels/CashgameCheckpointDeleteUrlModel.cs
using System.Web;
using Core.Classes;
using Core.Classes.Checkpoints;
using Web.Formatters;
using Web.Routing;

namespace Web.Models.UrlModels{

	public class CashgameCheckpointDeleteUrlModel : UrlModel{

		public CashgameCheckpointDeleteUrlModel(Homegame homegame, Cashgame cashgame, Player player, Checkpoint checkpoint)
			: base(GetUrl(homegame, cashgame, player, checkpoint)) {}

        private static string GetUrl(Homegame homegame, Cashgame cashgame, Player player, Checkpoint checkpoint)
        {
            var isoDate = cashgame.StartTime.HasValue ? UrlFormatter.FormatIsoDate(cashgame.StartTime.Value) : string.Empty;
			var encodedPlayerName = HttpUtility.UrlPathEncode(player.DisplayName);
			return string.Format(RouteFormats.CashgameCheckpointDelete, homegame.Slug, isoDate, encodedPlayerName, checkpoint.Id);
        }

	}

}